### PR TITLE
[MRG] MAINT: test docs in travis

### PR DESCRIPTION
**Pull request by GaelVaroquaux**
As discussed in #668 , we need to run the doctests too on travis.

### CODE REVIEW

**Comment by GaelVaroquaux** [comment]
This now finally works. Please review and merge

**Comment by lesteve** [comment]
For some reason, it seems like doctests are run from introduction.rst only. See https://travis-ci.org/nilearn/nilearn/jobs/71109899#L1044. Note I get the same thing if I run locally.

**Comment by AlexandreAbraham** [comment]
I replaced `doc/` in the `Makefile` by `find doc -name '*.rst'` and it works.

I started fixing the problems. However, is we decide not to allow dataset downloading, then I end up skipping all the doctests.

**Comment by GaelVaroquaux** [comment]
> I started fixing the problems. However, is we decide not to allow dataset
> downloading, then I end up skipping all the doctests.

Can you send in a pull request, so that we see how much damage there is?

**Comment by AlexandreAbraham** [comment]
#680

File: doc/introduction.rst [hunk 1]
```diff
@@ -508,7 +508,7 @@ Let's use all but the last 10 samples to train the SVC::
 
 and try predicting the labels on the left-out data::
 
-    >>> svc.predict(data[-10:])
+    >>> svc.predict(data[-10:])     # doctest: +SKIP
```
Comment (banilo) [inline doc/introduction.rst]: Isn't PEP8 to have only 2 spaces before #?

Comment (GaelVaroquaux) [inline doc/introduction.rst]: You're anal.

**Comment by lesteve** [comment]
Merging this one, thanks.

**Comment by GaelVaroquaux** [comment]
gr8

**Pull request merged**